lib/Containers/MerkleTree.cpp

```diff
@@ -1424,7 +1416,7 @@ void MerkleTree<Hasher, BranchingBits>::leftCombine(bool withShift) {
 
 template<typename Hasher, std::uint64_t const BranchingBits>
 void MerkleTree<Hasher, BranchingBits>::growRight(std::uint64_t key) {
-  std::unique_lock<std::shared_mutex> guard(_dataLock);
+  // not thread-safe, lock buffer from outside
```

Comment:
NIT
Maybe change the function name to growRightUnsafe or something like that?

Reply:
I considered that, but we already have several other functions with the same requirement which do not have this suffix, so I refrained from it to avoid confusion.